### DISCUSSION TITLE
GPTQ-for-LLaMa help

### DISCUSSION THREAD

**Discussion by MegaTechGuy** [oobabooga/text-generation-webui#2230] [Q&A]
Lads I need your help, how on earth do you get this to run?  I've spent a week trying to fix it but can get oobabooga run 4-bit models.

I've been stuck on this for a week and my brain 

>I'm a one click installer
>Installed on F: drive
>i7 4790K, 32GB Ram, GTX1080Ti 11Gb

Please help lads I'm going insane trying to make this work. 

![helppp](https://github.com/oobabooga/text-generation-webui/assets/134162102/f75b8c19-30e5-495b-8fab-1a9856206806)

**Comment by m-spangenberg** [comment]
Did you remember to:

1. go to the model tab and select the model, ignore the error
2. set the GPTQ parameters to: `wbits=4, model_type=llama, groupsize=128, pre_layer=11`
3. click save settings for this model
4. then click reload the model

**Reply by MegaTechGuy** [reply]
Thank you for suggesting that but I've already done that and all llama models dont seem to work even with the setting set.

**Reply by m-spangenberg** [reply]
That's unfortunate. And you followed the docs on [installing the necessary dependencies for GTPQ](https://github.com/oobabooga/text-generation-webui/blob/main/docs/GPTQ-models-(4-bit-mode).md) too?

If you're really left scratching your head, try running a 7B GGML model with GPU split but push the entire model to the GPU using `python3 server.py --cpu --n-gpu-layer 1000000000 --model-menu --chat`, might be worth a shot. This is just a guess, but if the GPU doesn't actually get used then you can narrow down the culprit to GPU drivers/CUDA.

**Reply by MegaTechGuy** [reply]
Tried a clean install and seem to be getting the same  GPTQ-for-LLaMa problem, REEEEEEEEEEEEEEE 
![failed to install](https://github.com/oobabooga/text-generation-webui/assets/134162102/519bb749-006a-4023-9afe-ff4d2bfe9f35)

**Reply by m-spangenberg** [reply]
It says you need `Microsoft's C++ Build Tools`

Scroll down to `Installing 4-bit LLaMA with text-generation-webui`: [how_to_install_llama_8bit_and_4bit](https://www.reddit.com/r/LocalLLaMA/comments/11o6o3f/how_to_install_llama_8bit_and_4bit/)

Hope this sorts you out.

**Comment by MegaTechGuy** [comment]
update error. 
![updateerror](https://github.com/oobabooga/text-generation-webui/assets/134162102/f53a0d70-451d-40d6-a886-89a03b31615b)
